spec/lang/step/intrinsics.md

```diff
@@ -40,7 +40,7 @@ impl<M: Memory> Machine<M> {
         if arguments.len() != 1 {
             throw_ub!("invalid number of arguments for `PointerExposeProvenance` intrinsic");
         }
-        let Value::Ptr(ptr) = arguments[0].0 else {
+        let Value::Ptr(Pointer { thin_pointer: ptr, ..}) = arguments[0].0 else {
```

Comment:
MIR currently requires this to be a thin pointer so I think we should do the same.

Reply:
Yes indeed, rustc gives a compilation error for `*const [u16] as usize`, noting "cast through a thin pointer first"